package/kernel/linux/modules/netfilter.mk

```diff
@@ -1147,6 +1147,30 @@ endef
 $(eval $(call KernelPackage,nft-bridge))
 
 
+define KernelPackage/nft-dup
+  SUBMENU:=$(NF_MENU)
+  TITLE:=Netfilter nf_tables dup support
+  DEPENDS:=+kmod-nft-core
+  KCONFIG:= \
+  CONFIG_NF_DUP_NETDEV \
+  CONFIG_NF_DUP_IPV4 \
+  CONFIG_NF_DUP_IPV6 \
+  CONFIG_NFT_DUP_NETDEV \
+  CONFIG_NFT_DUP_IPV4 \
+  CONFIG_NFT_DUP_IPV6
+  FILES:= \
+  $(LINUX_DIR)/net/netfilter/nf_dup_netdev.ko \
+  $(LINUX_DIR)/net/ipv4/netfilter/nf_dup_ipv4.ko \
+  $(LINUX_DIR)/net/ipv6/netfilter/nf_dup_ipv6.ko \
+  $(LINUX_DIR)/net/netfilter/nft_dup_netdev.ko \
+  $(LINUX_DIR)/net/ipv4/netfilter/nft_dup_ipv4.ko \
+  $(LINUX_DIR)/net/ipv6/netfilter/nft_dup_ipv6.ko
+  AUTOLOAD:=$(call AutoProbe,nf_dup_netdev nf_dup_ipv4 nf_dup_ipv6 nft_dup_netdev nft_dup_ipv4 nft_dup_ipv6)
```

Comment:
There is already a package with this file included:
https://github.com/openwrt/openwrt/blob/0dd5505a17959c0b3eabac2827031487444708fb/package/kernel/linux/modules/netfilter.mk#L1193
This would mean that we could not have both packages installed.

Will need to break the duplicated Kconfigs and files out into another package, and depend on that from both `nft-dup` and `nft-netdev`.
You will also need to address the `CONNTRACK` dependency: https://elixir.bootlin.com/linux/v5.15.111/source/net/ipv4/netfilter/Kconfig#L38

---
Probably best to wait for a second opinion before starting these next parts:

Could also define a Make variable for the dup Kconfigs, and files: https://github.com/openwrt/openwrt/blob/0dd5505a17959c0b3eabac2827031487444708fb/include/netfilter.mk, but given the mix in the package Makefile, it seems like that may no longer be needed.

`ipt-tee` does include the `NF_DUP_IPV*` symbols (being an iptables package, not worried about that conflict), but you could break those out into their own Make variables (in the include file), and packages.
https://github.com/openwrt/openwrt/blob/0dd5505a17959c0b3eabac2827031487444708fb/package/kernel/linux/modules/netfilter.mk#L717
https://github.com/openwrt/openwrt/blob/0dd5505a17959c0b3eabac2827031487444708fb/include/netfilter.mk#L257

Reply:
Thx for comment. I didn't know that this will make conflict so as you suggested I've splitted this into separate packages.

About second part have you any docs/links that suggests how to do that? Because as I'm looking into that file I can't figure out how variables works there and I'm not very familiar with makefile syntax,

Reply:
About using the netfilter include variables:
No _need_ to use them (unless a committer wants you to).
To learn about it:
`nf_add` is a Makefile function / macro which appends to variables (name given by a parameter) when called: `$(call nf_add, variable-name, Kconfig, module-filename-no-extension, optional-kernel-version-condition)` gives `KCONFIG_variable-name`, and `variable-name-$(Kconfig-state)`, so you get `-m` suffix for modules: `variable-name-m` (a list).
These variables are then refered to in the netfilter package Makefile to get the paths of `.ko` files, and the modprobe names using the `-m` suffix variable, so that the packages do not miss files if the Kconfig is built-in to kernel (that would create a `-y` suffix variable).
Good example usage in a previous commit: https://github.com/openwrt/openwrt/commit/144fa4d4e7c00df370924751f58781689cd333e7